From a4b9e9099a3e5d5fc3837368e8ee08c26485206c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 14 Sep 2021 18:05:23 +0300
Subject: [PATCH] Revert "[SW-0] gaudi2: arc: fix the sram base"

The documentation is misleading stating the region is SRAM, while in
fact it is not pure SRAM, but includes also a reserved area which is not
populated with SRAM. Thus the actual SRAM array starts at offset of 16MB
from the SRAM region start.
And thus the correct base for actual SRAM array start is
SRAM region base (0x10000000) + reserved area size (16MB) = 0x11000000.
Revert the fix as it is wrong and add a comment explaining the correct
SRAM base.

This reverts commit dfc6fee7d19f9f861decaf503345ac836695ff0d.

Change-Id: Ibb3b12460c579bc86f4f2fa69b0c57d08c69182b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index b9ddc25..ec5c655 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -17,7 +17,13 @@
 #define HL_AXI_GENERAL_BASE	0x1000007FF0000000ULL
 #define HL_SP_SRAM_BASE		0x1000007FF7FE0000ULL
 #define HL_SP_SRAM_SIZE		0x10000
-#define SDRAM_BASE		0x10000000
+/*
+ * The SRAM region in ARC address extension is mapped to 0x10000000, but the
+ * first 16MB of that region are not populated with SRAM H/W, so the actual
+ * SRAM array starts at offset of 16MB from the region start and is of 48MB size
+ * That makes the SRAM reside in range of 0x11000000 - 0x13FFFFFF (inclusive).
+ */
+#define SDRAM_BASE		0x11000000
 #define GEN_REGION_BASE		0x20000000
 #define PCIE_FW_SRAM_BASE	(GEN_REGION_BASE + 0x7FF0000)
 #define FW_SRAM_BASE		(GEN_REGION_BASE + 0x7FD0000)
